templates/resources/vault_secrets_secret.md.tmpl

```diff
@@ -0,0 +1,18 @@
+---
+page_title: "{{.Type}} {{.Name}} - {{.ProviderName}}"
+subcategory: "HCP Vault Secrets"
+description: |-
+{{ .Description | plainmarkdown | trimspace | prefixlines "  " }}
+---
+
+# {{.Name}} ({{.Type}})
+
+-> **Note:** Please treat your state file as sensitive when using this resource.
```

Comment:
This is standard practice for all Terraform state files when using resources that have attributes that are marked sensitive. I don't think the disclaimer is necessary, but if still desired, I'd recommend copying the one used by the `vault_cluster_admin_token` resource (includes a link to best practices).

Reply:
This is a decision that was made from the Vault Secrets team. I think we're just being extra cautious with folks who choose to use this particular resource.